### PR TITLE
pythonPackages.python-forecastio: init at 1.4.0

### PR DESCRIPTION
**Pull request by makefu**
###### Motivation for this change

add the python package to get the darksky home-assistant module to work.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [NA] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`) - works with home-assistant
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by ryantm** [comment]
@GrahamcOfBorg build python2.pkgs.python-forecastio python3.pkgs.python-forecastio

File: pkgs/servers/home-assistant/component-packages.nix [hunk 1]
```diff
@@ -766,7 +766,7 @@
     "sensor.cups" = ps: with ps; [ pycups ];
     "sensor.currencylayer" = ps: with ps; [  ];
     "sensor.daikin" = ps: with ps; [  ];
-    "sensor.darksky" = ps: with ps; [  ];
+    "sensor.darksky" = ps: with ps; [ python-forecastio ];
```
Comment (ryantm) [inline pkgs/servers/home-assistant/component-packages.nix]: Was this generated by `parse-requirements.py`? The top of the file says "do not edit".

Comment (makefu) [reply pkgs/servers/home-assistant/component-packages.nix]: oh oh :D i will look into it

Comment (makefu) [reply pkgs/servers/home-assistant/component-packages.nix]: I ran the command, however it was unable to detect the new dependency. now sure what to touch to make it work.
The package is called python-forecastio and will be installed with `pip install python-forecastio`. (see https://github.com/ZeevG/python-forecast.io )
can you check the parse-requirements.py package what went wrong there?

Comment (ryantm) [reply pkgs/servers/home-assistant/component-packages.nix]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/sensor/darksky.py#L22 requires it

Comment (makefu) [reply pkgs/servers/home-assistant/component-packages.nix]: exactly, but how can i get the `parse-requirements.py` to find this requirement and add it automatically to the list of components?

Comment (ryantm) [reply pkgs/servers/home-assistant/component-packages.nix]: cc @peterhoeg @dotlambda

Comment (FRidh) [inline pkgs/servers/home-assistant/component-packages.nix]: This needs to be a separate commit.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (makefu) [reply pkgs/servers/home-assistant/component-packages.nix]: as discussed above, this should be done somehow automatically by the update script but it doesn't for some reason

Comment (peterhoeg) [reply pkgs/servers/home-assistant/component-packages.nix]: I have just tried running the ```parse-requirements.py``` script here and it works perfectly fine. In addition to updating the darksky ```sensor``` component it also updates the ```weather``` component as expected. What happens when you run it?

As mentioned by @FRidh that needs to go into a separate commit.

File: pkgs/development/python-modules/python-forecastio/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 = "0m6lf4a46pnwm5xg9dnmwslwzrpnj6d9agw570grciivbvb1ji0l";
+
+  };
+
+  checkInputs = [ nose coverage responses ];
+
+  propagatedBuildInputs = [ requests responses ];
+
+  checkPhase = ''
+    nosetests --with-coverage
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-forecastio/default.nix]: You should be able to just drop `--with-coverage` and the `coverage` dependency above.

File: pkgs/development/python-modules/python-forecastio/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  pname = "python-forecastio";
+  version = "1.4.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0m6lf4a46pnwm5xg9dnmwslwzrpnj6d9agw570grciivbvb1ji0l";
+
+  };
+
+  checkInputs = [ nose coverage responses ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-forecastio/default.nix]: no need for `responses` here as it's in `propagatedBuildInputs`

File: pkgs/servers/home-assistant/component-packages.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/servers/home-assistant/component-packages.nix]: `parse-requirements.py` works for me as well. You should however rebase onto latest master, where Home Assistant was updated to 0.72.0.

**Comment by makefu** [comment]
i've added the changes requested, rebased onto latest master and i've put the regenerated component-packages into a separate commit.

**Comment by peterhoeg** [comment]
There are 2 changes to the components file but one of them is in the wrong commit.

**Comment by makefu** [comment]
@peterhoeg you are right, i fixed that

**Comment by dotlambda** [comment]
The file is still changed by the first commit.

**Review by FRidh — APPROVED** [review]

**Review by FRidh — CHANGES_REQUESTED** [review]
as @dotlambda mentioned

**Comment by makefu** [comment]
... sorry for the fuckup, this is what happens when you try to manually revert changes in a commit instead of knowing the correct git command.
it should be ok now .

**Comment by dotlambda** [comment]
Thanks @makefu! That Dark Sky API seems to be interesting indeed.
